src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/WeakReferenceCache.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Threading;
+
+namespace System.Windows.Media.Imaging;
+
+/// <summary>
+/// Provides keyed-caching for different objects stored as weak references.
+/// </summary>
+/// <remarks>The cache operations are thread-safe.</remarks>
+internal sealed class WeakReferenceCache<K, V> where K : notnull
+                                               where V : class
```

Comment:
Nit, personally I put `where` on separate lines. It seems like it's the standard in this repo and in the WinForms repo, especially when there's more than 1 type parameter.

Something like this:
```suggestion
internal sealed class WeakReferenceCache<K, V>
    where K : notnull
    where V : class
```

Reply:
Didn't really check how 2+ are done, adjusted, don't really have a preference.